src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,520 @@
+
+    auto getMaxKnotTime = [](const UnicyclePlannerInput& input) -> double {
+        double maxKnotTime = 0.0;
+
+        for (const auto& knot : input.knots)
+        {
+            if (knot.time > maxKnotTime)
+                maxKnotTime = knot.time;
+        }
+
+        return maxKnotTime;
+    };
+
+    if (input.tf < getMaxKnotTime(input))
+    {
+        log()->error("{} The input contains a knot whose time is over the planner horizon.",
+                     logPrefix);
+        return false;
+    }
+
+    m_pImpl->planner->clearDesiredTrajectory();
+
+    for (const auto& knot : input.knots)
+    {
+        auto position = iDynTree::Vector2();
+        position[0] = knot.x;
+        position[1] = knot.y;
+
+        auto velocity = iDynTree::Vector2();
+        velocity[0] = knot.dx;
+        velocity[1] = knot.dy;
+
+        if (!m_pImpl->planner->addDesiredTrajectoryPoint(knot.time, position, velocity))
+        {
+            m_pImpl->planner->clearDesiredTrajectory();
+            log()->error("{} Failed to insert knot in the Unicycle planner.", logPrefix);
+            return false;
+        }
+    }
+
+    m_pImpl->horizon.t0 = input.t0;
+    m_pImpl->horizon.tf = input.tf;
+
+    return true;
+}
+
+bool Planners::UnicyclePlanner::advance()
+{
+    constexpr auto logPrefix = "[UnicyclePlanner::advance]";
+
+    if (!m_pImpl->planner)
+    {
+        log()->error("{} The Unicycle planner has never been initialized.", logPrefix);
+        return false;
+    }
+
+    m_pImpl->left = std::make_shared<FootPrint>();
+    m_pImpl->left->setFootName("left");
+
+    m_pImpl->right = std::make_shared<FootPrint>();
+    m_pImpl->right->setFootName("right");
+
+    if (!m_pImpl->planner->computeNewSteps(m_pImpl->left,
+                                           m_pImpl->right,
+                                           m_pImpl->horizon.t0,
+                                           m_pImpl->horizon.tf))
+    {
+        m_pImpl->left = nullptr;
+        m_pImpl->right = nullptr;
+        log()->error("{} Failed to compute new steps.", logPrefix);
+        return false;
+    }
+
+    auto convertStep
+        = [&](const ::Step& step, const ::Step& nextStep) -> const Contacts::PlannedContact {
+        Contacts::PlannedContact contact;
+        contact.name = step.footName;
+        contact.activationTime = step.impactTime;
+
+        contact.pose.quat(Eigen::AngleAxisd(step.angle, Eigen::Vector3d::UnitZ()));
+        contact.pose.translation({step.position[0], step.position[1], 0.0});
+
+        // Note: we use the impact time to compute the duration of the step.
+        //       If the step is the last one, a default 5 seconds duration is used.
+        const double duration = nextStep.impactTime - step.impactTime;
+        assert(duration >= 0);
+        contact.deactivationTime
+            = duration > 0.0 ? contact.activationTime + duration * m_pImpl->gait.stancePhaseRatio
+                             : contact.activationTime + 5.0;
```

Comment:
Actually, the duration set to zero makes sense only if it is the last step. If it is not the last step, it is an error. I am not very convinced in using a random hard-coded value.

In case it is the last step, in principle, there is no deactivation time. So we could decide:
- to use the horizon of the planner
- to use the numeric limit
- to use a negative value.
- to add a flag to ``PlannedContact``

Any opinion on this @GiulioRomualdi @traversaro @prashanthr05?

In any case, this should be properly documented.

Reply:
I do not have any preference w.r.t. to the options listed by @S-Dafarra .

Reply:
This was indeed a TODO point. In the first implementation I was using option: _to use the horizon of the planner_. However, this is not strict since the time of the generated footsteps can be higher than the nominal horizon. I'm open to other ideas.

Reply:
From the options listed by @S-Dafarra, I would either go for a negative value (undecided contact time) or to use a numeric limit (wait infinitely, atleast according to the platform we use it in).

Reply:
I like the idea of using `std::numeric_limit<double>::max()`

Reply:
Done in ebb2d11

Reply:
It is a bit odd when printed, but it makes sense to me:

```
Contact (name = right, pose = SE3 (position = [0.97101, 0.14843, 0], quaternion = [0, 0, 0.165852, 0.986151]), activation_time = 21.730000, deactivation_time = 179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000, type = 0)
```

Reply:
Probably printing in scientific notation can help in this case.

Reply:
Done in d6e7ed3, now it prints:

```
Contact (name = right, pose = SE3 (position = [0.97101, 0.14843, 0], quaternion = [0, 0, 0.165852, 0.986151]), activation_time = 1.234568, deactivation_time = 1.797693e+308, type = 0)
```